### ISSUE TITLE
Argument list too long error for ldd run in debbuild

### ISSUE THREAD

**Issue by Conan-Kudo** [ascherer/debbuild#76]
In one of my packages, the ldd run in debbuild is apparently so long that it fails:

With debbuild 17.3.0:
```
[   61s] Can't exec "/bin/sh": Argument list too long at /usr/bin/debbuild line 1567.
```
In current master, that is line 1584:

```perl
  my @reqlist = grep { ! m|^/| } qx ( LANG=C $specglobals{__ldd} $binlist );
```

It seems like it might be a better idea to iterate on each file and then deduplicate the resulting list that is pushed to the control file's `Depends` line.

The issue does still occur on current master.

**Comment by ascherer** [comment]
What's the approximate size limit for the argument list. Maybe it can be split into suitable chunks.

**Comment by ascherer** [comment]
Fixed in 17.4.1.

**Comment by ascherer** [comment]
A shell commandline is too long, no matter what program you call.

**Comment by ascherer** [comment]
Piping to STDOUT would be possible, but we need the results from STDIN. Solve this with the help of a temporary file.

**Comment by ascherer** [comment]
Fixed in release 17.4.2.
